docs/gitbook/usage/metrics.md

```diff
@@ -401,3 +401,78 @@ Reference the template in the canary analysis:
           max: 5
         interval: 1m
 ```
+
+## Graphite
+
+You can create custom metric checks using the Graphite provider.
+
+Graphite template example:
+
+```yaml
+apiVersion: flagger.app/v1beta1
+kind: MetricTemplate
+metadata:
+  name: graphite-request-success-rate
+spec:
+  provider:
+    type: graphite
+    address: http://graphite.monitoring
+  query: |
+    target=summarize(
+      asPercent(
+        sumSeries(
+          stats.timers.httpServerRequests.exception.*.method.*.outcome.{CLIENT_ERROR,INFORMATIONAL,REDIRECTION,SUCCESS}.status.*.uri.*.count
```

Comment:
Was this tested? I haven't used Graphite in ages but this query seem wrong to me, also it doesn't filter the canary pods.

Reply:
@stefanprodan: We have updated the query to include the `{{target}}` parameter. One question we do have, is the query intended to consider metrics for both the primary and canary pods, or just the canary pods?

As for the query's correctness, we went ahead and set up a demo Spring Boot app that ships metrics to graphite to test this out. You can find the example here: https://github.com/nmische/spring-boot-graphite. Based on some testing with that application we believe this query correctly calculates the request success rate using the default metrics generated by Spring Boot (with an additional tag `app` configured to identify the application).

Reply:
The builtin checks are for the canary pods only, my assumption is that users don't want a canary analysis to fail if the older version (primary) is running into errors, imagine you have some bug in production, if you look at both versions you'll never manage to deliver a fix.

Assuming that your app uses the Kubernetes Deployment name when it reports to Graphite, then the current query is ok, as the primary app name will be `{{target}}-primary`.